File: Nanopore_HEK293/scripts/get_features.R
args <- commandArgs(trailingOnly = TRUE)
#Read preprocessed JACUSA2 output
print("Rebuild matrix of features ...")
Exp1 <-
  read.delim(paste0(args[1], "/data_reformat.txt"),
             as.is = T,
             header = F)

colnames(Exp1) = c(
  "ID",
  "contig",
  "position",
  "call2.score",
  "deletion.score",
  "insertion.score",
  "Base",
  "Anchor",
  "Strand"
)
if (dir.exists(dirname(args[2])) == FALSE) {
  dir.create(dirname(args[2]))
}

Call2 <- tapply(Exp1$call2.score, list(Exp1$ID, Exp1$Anchor), sum)
Call2[is.na(Call2)] <- 0
colnames(Call2) <- paste0("Exp1Call2Score_", colnames(Call2))

Deletion <- tapply(Exp1$deletion.score, list(Exp1$ID, Exp1$Anchor), sum)
Deletion[is.na(Deletion)] <- 0
colnames(Deletion) <- paste0("Exp1DeletionScore_", colnames(Deletion))

Insertion <-
  tapply(Exp1$insertion.score, list(Exp1$ID, Exp1$Anchor), sum)
Insertion[is.na(Insertion)] <- 0
colnames(Insertion) <-
  paste0("Exp1InsertionScore_", colnames(Insertion))

BigTable <-
  merge(
    data.frame(ID = rownames(Call2), Call2),
    data.frame(ID = rownames(Deletion), Deletion),
    by.x = 1,
    by.y = 1
  )
BigTable <-
  merge(BigTable,
        data.frame(ID = rownames(Insertion), Insertion),
        by.x = 1,
        by.y = 1)


motif <-
  read.table(paste0(args[1], "/checkMotif_reformat.txt"),
             as.is = T,
             header = F)
motif[, 1] <- gsub("-", "_", motif[, 1])
motif[, 1] <- gsub("\\(\\_\\)", ":-", motif[, 1])
motif[, 1] <- gsub("\\(\\+\\)", ":\\+", motif[, 1])

BigTable <- merge(BigTable, motif, by.x = 1, by.y = 1)


colnames(BigTable)[ncol(BigTable)] <- "Motif"
rownames(BigTable) <- BigTable[, 1]
BigTable <- BigTable[, -1]

BigTable$DRACH <- rep(0, nrow(BigTable))
BigTable$DRACH[grep("[AGT][AG]AC[ACT]", BigTable$Motif)] <- 1
saveRDS(BigTable, file = args[2])

File: Nanopore_HEK293/scripts/get_features.pl

while(<>)
{
	chomp;
	my @tmp = split(/\t+/,$_);
	my $offset=$#tmp-15;

	my ($contig, $ref_position, $callScore, $Indel, $base, $site, $strand) = ($tmp[0],$tmp[1]."_".$tmp[2],$tmp[10],0,$tmp[$offset+15], $tmp[7], $tmp[11]);
	my($deletionScore, $insertionScore) = (0,0);
	if(/deletion\_score\=([\d\.E-]+)/)
	{ $deletionScore=$1;}

	if(/insertion\_score\=([\d\.E-]+)/)
	{ $insertionScore=$1;}

	$site=$site-$tmp[1]+1 if($strand eq '+');
        $site=$tmp[2]-$site if($strand eq '-');
							  
	printf("%s\t%s\t%s\t%.2f\t%.2f\t%.2f\t%s\t%d\t%s\n",$contig.":".$ref_position.":".$strand, $contig, $ref_position, $callScore, $deletionScore, $insertionScore, $base, $site, $strand);
}

